var/spack/repos/builtin/packages/libpressio/package.py

```diff
@@ -427,10 +429,12 @@ def test(self):
             "-DCMAKE_PREFIX_PATH={};{}".format(self.spec["libstdcompat"].prefix, self.prefix)
         )
 
-        self.run_test("cmake", args, purpose="cmake configuration works")
+        cmake = which("cmake")
+        cmake(*args)
+        cmake("--build", ".")
 
-        # this works for cmake@3.14: which is required for this package
-        args = ["--build", "."]
-        self.run_test("cmake", args, purpose="cmake builds works")
+        exe = which("./pressio_smoke_tests")
+        out = exe()
```

Comment:
The output-related arguments are missing.  See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#reading-expected-output-from-a-file for the missing arguments.